pkg/ssh/client.go

```diff
@@ -67,13 +73,34 @@ func (c *Client) Connect(ctx context.Context) error {
 		HostKeyAlgorithms: []string{ssh.KeyAlgoED25519},
 	}
 
-	conn, chans, reqs, err := ssh.NewClientConn(tcpConn, tcpConn.RemoteAddr().String(), conf)
-	if err != nil {
-		return err
-	}
+	respCh := make(chan connResp)
 
-	c.conn, c.client = conn, ssh.NewClient(conn, chans, reqs)
-	return nil
+	// ssh.NewClientConn doesn't take a context, so we need to handle cancelation on our end
+	go func() {
```

Comment:
it'd be far simpler instead of launching this in the background, and since you already have a reference to the underlying connection, to just close it on context cancellation instead of this.